### ISSUE TITLE
Sampling, Crowd-Sourcing & Reliability - Fundamentals

### ISSUE THREAD

**Issue by jamesallenevans** [Computational-Content-Analysis-2020/Readings-Responses-Spring#10]
Post questions about the following fundamentals reading here:

Dawid, A. P., and Skene, A. M. 1979. [“Maximum Likelihood Estimation of Observer Error-rates using the EM Algorithm.”](https://www.jstor.org/stable/pdf/2346806.pdf) Applied Statistics 28(1): 20-28.

Mechanical Turk tutorial on Canvas (essential for performing the assignment if you haven't used it before.)

**Comment by wanitchayap** [comment]
> While the vast majority of the final estimates of the T's remain virtually unaltered as different local maxima are discovered, some T estimates change markedly indicating the "reclassification" of one or two patients. A similar feature is noticed when the algorithm is repeated following the removal of cases. Thus it is very easy to establish which patients are well classified and for which patients the consensus is more uncertain.

This is a very interesting application of MLE that I have never known before! Can this be used with other applications than tasks involving several annotators in text analysis?  If so, can you give some examples in the context of content analysis?

**Comment by ihsiehchi** [comment]
I was considering possible applications of the EM algorithm to the sampling methods discussed in Krippendorf (2004), one promising candidate that emerged in my mind was the relevance sampling method. In particular, we employ the EM algorithm to evaluate how accurately an individual or an algorithm is assigning texts the binary relevance indicator value. However, pertinent to the discussion of limitations of Dawid and Skene (1979), relevance sampling may constitute a case of violation in the sense that "true responses" do not exist, or the true response can not be determined as easily as the method of the wisdom of the crowd (or even with the wisdom of experts) as described in DS (1979). More generally, beyond the EM algorithm, how do we evaluate the validity of relevance sampling?

**Comment by nwrim** [comment]
I had only seen EM algorithm in the context of mixture models until now, so this was a really interesting reading so far! The authors mention that EM algorithm has the problem of finding the local optima instead of global optima, and suggests that we should try multiple starting points. This gives me two questions:

1) Is there any systematic way to choose the various starting points? For example, if we choose two starting points that fall into the same local optima by bad luck, then we would think that the local optima is the global optima. Are there any ways to prevent or decrease the probability of this happening? In addition, are there any rules to choose the number of starting points to try?

2) It feels like that it would be computational inefficient to run EM algorithm on a large corpus, especially when we have to try a lot of starting points. Since this article was written quite long ago, I wonder if there are more algorithms that do the job more efficiently or accurately. As a coding presenter for this week, I looked at several annotation models implemented in `pyanno` (which you wrote!) - what are some advantages/disadvantages of EM algorithm compared to these approaches? How should we choose what approaches/models to use?

**Comment by timqzhang** [comment]
It is a good approach to inspire people to think a lot on the observer error-rates. From what are mentioned in the paper, I here have following questions:

1. It seems that the paper is quite unsure about the accuracy of MLE estimates, as mentioned in the discussion section: 

> Little is known at present about the accuracy of the estimates.

I understand that the main purpose for this paper may lie more in the EM algorithm, but from the perspective of application, EM and MLE should be considered as a combo. Otherwise, it is less meaningful to develop one algorithm to make sure the estimates converge in MLE, while uncertain about how accurate the MLE estimates are. As it is a pager published in 1978, I totally understand the difficulties for that time to go further, and I am willing to see whether and how such estimates are tested and applied to verify its robustness? 

2. The EM algorithm is quite based on the number of observer, namely k, while it is hard to keep accurate tracing of these records, since when the data gets much larger, a complex network could be constructed, and it makes the identification of relations and collections even harder. Is there any efficient data tracing to make sure that the raw data is not missing something and could be trusted? Also, I wonder the application of this algorithm, especially in the content analysis, which may arouse another question on whether this approach is still competitive, as it is developed in last century, compared with many new approaches nowadays?

**Comment by WMhYang** [comment]
1. The MLE requires both maximum likelihood estimate and missing data estimate to converge. From my experience with MLE, sometimes it can be really hard to guarantee convergence, where initial values and sample properties can matter. How could we deal with our samples and choose intial guesses to make sure convergence?
2. It is mentioned in the paper that the MLE has to estimate a large set of parameters, so it has few advantages except for tractability and feasibility. To alleviate this concern, the authors were also using smaller numbers of parameters to do the estimation. Given that in this time and age we have more techniques to obtain big data, will high parameterized models become more attractive to researchers?

**Comment by pdiazm** [comment]
It seems like there are a lot of caveats to the EM method, which makes me hesitant about its usefulness. One that I found particularly striking is the assumption of no patient-by-clinician interaction, which (if I understood correctly) doesn't allow the quality of the response to be related to the ability of the interviewer. I don't think there is a single situation in which a response is not dependent on the question. Am I understanding this correctly or does the author mean something different?

**Comment by iarakshana** [comment]
As @wanitchayap had mentioned, it would be great to know some more applications of this method because I am not able to see how it would be applied in content analysis. I guess it would be used to estimate the true responses of the texts we're analyzing in some way? Also, it seems like in this case, measurement error can be extracted based on the anomalies of the patient information but I feel like when automating this extraction computationally, we might be classifying  words that were intended to be used (for example some technical jargon in the sciences maybe) as error just due to the use frequency?

**Comment by linghui-wu** [comment]
First, I agree with the first point of @WMhYang on the convergence requirements of MLE. Moreover, I want to add that the convergence of data plays an indispensable role in not only  MLE but also the EM algorithm, especially when consensus is not apparent in the observations.

Second, I am also wondering about the real-life application of this methodology, particularly when we cannot exhaust all kinds of "errors" and figure out the "true answers". What if we cannot distinguish between erroneous responses and valid responses at the first stage of the experiment?

**Comment by harryx113** [comment]
As a two-step iterative approach, EM model's computational availability partially depends on the quality of the missing data. My question is, when the missing values are in a sequence of events and  the number of values grows quickly, how do we ensure a reliable iteration?

**Comment by tianyueniu** [comment]
Relating to previous comments, what are some examples on the application of EM and MLE in content analysis? Can we use it for parsing grammar structures?

**Comment by timhannifan** [comment]
What does "missing data" mean in the context of text data? I understand the concept in terms of categorical, numeric, or boolean features, but how does one impute an n-gram, domain name, or IP address?

With regard to evaluating the impact of individual observers, the EM algo is interesting, but it's not clear how it helps improve estimates when the true response/outcome is never known. For example, consider the case of mechanical turks labeling news articles as disinformation or not: the true outcome (or label) is never known, just the best guess of our hired help.

Lastly, assumption two (no patient/physician interaction) is laughable in the context of anesthesiology, and likely many others.

**Comment by Lesopil** [comment]
Perhaps this is a dumb question. I understand how the MTurk data collection may relate to content analysis, especially in light of the "Human language reveals a universal positivity bias" reading. This might be used to aggregate other interpretations of the meanings/vibes of words. However, I have difficulty imagining how we could apply the Dawid and Skene reading to this. It seems to me, that there will not be a right or wrong answer when assigning meaning to a word. How can we adequately prepare for this potential lack of a right or wrong answer among the responses?

**Comment by DSharm** [comment]
I agree with some of the above comments about the implausibility of assumption 2 (that the interviewer / clinician does not matter in obtaining better or worse answers from the interviewee / patient). Given that this assumption is likely to be violated, how would we expect this to effect our estimates? As a corrolary to @wanitchayap's question, if this method can be used in tasks involving several annotators, but we might expect some annotators to be better / more careful than others, can we account for this variability?

**Comment by Yilun0221** [comment]
I really like this topic which is very meaningful in our daily life. As for the convergence problem metioned above, I think the researchers can set a limit to the steps of running the algorithm. In addition, in the beginning of this paper, it says that there may be multiple reasons for observer rates and they listed some of them. I wonder how do they get an unbiased sample from the target population to 'evenly' reflect possible reasons and maybe some potential but unknown ones?

**Comment by jsgenan** [comment]
The algorithm is very friendly set in the patient-physician relationship, which makes it very clear what is the benefit and defficient of the procedures. I was wondering, however, how can we generalize the algorithm? For example, I think the assumptions require that j~k relationship is independent of observer i. What could be other necessary conditions to apply this algorithm?

**Comment by lyl010** [comment]
The EM algorithm applied to conduct MLE seems to be a powerful tool even if we do not know the 'true response' from respondents. But my question is why, p(Tij|data) is important? Another question is: Is EM updating estimation of missing data and MLE in a way as if this algorithm is based on the assumption that missing data are of the same pattern with the seen data?

**Comment by liu431** [comment]
I am wondering what are some new improvements to speed up the EM algorithm?

**Comment by bazirou** [comment]
Good work! I am wondering whether other methods can be used to estimate the MLE.
